continuous-testing/starter-kits/README.md

```diff
@@ -2,6 +2,403 @@
+
+3. Install pre-commit hook
+```bash
+pre-commit install
+```
+This command reads the `.pre-commit-config.yaml` file and installs the pre-commit hook in your local git repository.
+After this, you can see a `.git/hooks/pre-commit` file is created in your local git repository.
+
+4. Make sure you have the baseline file `.secrets.baseline` in your local git repository
+
+5. Commit your changes
+
+Now, you can commit your changes as usual. If any **new secrets** are detected, the commit will be prevented and the secrets will be reported.
+
+For example,
+
+<img width="559" alt="Screen Shot 2023-04-20 at 7 32 10 AM" src="https://user-images.githubusercontent.com/92573736/233398613-6e189322-4d97-47c3-b3ba-bd700a716cf6.png">
+
+
+> **Note**: pre-commit hook block commit by comparing new secrets with the results in `.secrets.baseline` file. If you want to add new secret results, you need to update `.secrets.baseline` file by re-running the scan command and generate a new baseline file.
+>
+> You can create an empty result baseline file by running this command at a directory without secrets.
+
+#### Layer 3: Server-side Push to GitHub.com
+The final layer of our solution is a server-side pre-commit scan powered by [GitHub Action](https://github.com/features/actions). This scan is triggered whenever a developer pushes to a branch or creates a pull request. If the scan detects any new secrets, it can generate a not detailed report compared to layer 2 (for security concern), email to the developer, and report a status check to GitHub. The status check will prevent the developer from merging the pull request or pushing to the **protected** branch. This layer protects the protected branch from being polluted by secrets, but secrets can still be pushed to other branches.
+
+```mermaid
+sequenceDiagram
+    participant User as Developer
+    participant GH as GitHub
+    participant Workflow as detect-secrets.yaml
+    participant GA as GitHub Action
+    participant DS as Detect-Secrets
+
+    Note over User,GH: Developer creates pull request or pushes to branch
+    User->>+GH: Creates pull request / pushes to branch
+    GH->>+Workflow: Triggers GitHub Action workflow
+    Workflow->>GA: Sets up and runs Detect-Secrets scan
+    GA->>DS: Requests secret scan
+    DS->>DS: Scans repository for secrets
+    alt Secrets Detected
+        DS-->>GA: Returns detected secrets
+        GA-->>GH: Fails status check
+        GH-->>User: Prevents merge / push & sends email notification
+    else No Secrets Detected
+        DS-->>GA: Returns clean result
+        GA-->>GH: Passes status check
+        GH-->>User: Allows merge / push
+    end
+
+```
+Starter Kit:
+1. Create a workflow file `detect-secrets.yaml` in `.github/workflows` directory from your repository root.
```

Comment:
After running this on a fresh repository on GitHub, I received the following in the GH Actions workflow log (see trace [here](https://github.com/riverma/test-repo/actions/runs/5139169327/jobs/9249315949)):

```
Run # scripts to scan repository for new secrets
cp: cannot stat '.secrets.baseline': No such file or directory
Error: Process completed with exit code 1.
```

Is there a way you can check for and create a blank `.secrets.baseline` file if being run for the first time on a repo that has never used detect secrets before?

Reply:
try `touch .secrets.baseline` which will ensure it's always there.

Reply:
> try `touch .secrets.baseline && <additional_command>` which will ensure it's always there.

Yep, I have a similar idea. But due to the format requirement of the baseline file (detect-secrets version can be different), my new solution is to generate the blank baseline file (but with the correct config) by scanning the result on an empty folder.
```yaml
      - name: Create a blank .secrets.baseline (but with correct configuration) if it does not exist
        run: |
          if [ ! -f .secrets.baseline ]; then
            # The generated baseline file will only be available temporarily on the GitHub side and will not appear on the user's local files
            # Scanning an empty folder ensures no initial secrets are included
            echo "⚠️ No existing .secrets.baseline file detected. Creating a new blank baseline file."
            mkdir empty-dir
            detect-secrets scan empty-dir > .secrets.baseline
            echo "✅ Blank .secrets.baseline file created successfully."
            rm -r empty-dir
          else
            echo "✅ Existing .secrets.baseline file detected. No new baseline file will be created."
          fi
```

Reply:
Touch does exactly that. If it's not there, it creates an empty file. 

You may want to use `>>` so if a file exists it doesn't replace its contents with an empty file.

Reply:
> You may want to use `>>` so if a file exists it doesn't replace its contents with an empty file.

I have a clarification. 

To run detect-secrets successfully, the baseline file should not be entirely empty but should contain some information, even if no secrets are detected. Here's an example of the required format for the baseline file:

<img width="488" alt="image" src="https://github.com/NASA-AMMOS/slim/assets/92573736/2d87794c-10e3-40aa-ba5b-52bbc9bde670">

And, this format of the content generated can vary based on the version of detect-secrets. So I use this command to help with the compatibility
`detect-secrets scan empty-dir > .secrets.baseline`

My solution uses `if-else` approach to determine whether we need to create the "initial" baseline file. And even though the file got created by the GitHub Action server, the change is not committed, so it will not affect out project code.

Reply:
Hi @perryzjc - thanks for getting the GitHub Action going. Working now in my [test run](https://github.com/riverma/test-repo/actions/runs/5224805784/jobs/9433474806)! Though please see two suggestions below: 

Suggestions:
1. The sample workflow file mentioned in your Usage section does not work as-is for me. I received an invalid workflow file exception (see [here](https://github.com/riverma/test-repo/actions/runs/5224837125) and [here](https://github.com/riverma/test-repo/actions/runs/5224849234)). I had to remove the "description" and the "branding" parameters to get it to work. Can you resolve this?
2. I'm not clear on the "Use latest version" button of your [GH action page](https://github.com/marketplace/actions/detect-secrets-action) and how your action will support inline use within the "steps" section of a GitHub workflow file. Seems like the way you have it configured, your GitHub action can only work as a single workflow file, and not within a workflow file that contains multiple steps. Might be good to clarify that in the description  of your GitHub action.

Reply:
After chatting with @perryzjc - let's go forth with reverting to the previous GitHub Action strategy, namely: embedding the Workflow file contents inline into this documentation for users to copy/paster. The GH marketplace strategy is complicated and will require more investigation.

Reply:
@riverma I just made some [updates](https://github.com/NASA-AMMOS/slim/pull/95/commits/d27620858dfaa0a34b6a04273259ca2e96007753) based on the conversation.